packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1980,3 +1982,61 @@ typedef RoutePageBuilder = Widget Function(BuildContext context, Animation<doubl
 ///
 /// See [ModalRoute.buildTransitions] for complete definition of the parameters.
 typedef RouteTransitionsBuilder = Widget Function(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child);
+
+/// A widget that steals focus from other focusable nodes.
+///
+/// This is used to allow clicking outside of a textfield or other
+/// editable text to "de-select" the text field without needing to
+/// jump to another focusable widget.
+class _FocusTrap extends StatefulWidget {
+  /// Create a new [FocusTrap] widget.
+  const _FocusTrap({
+    required this.child,
+    this.focusNode,
+    Key? key
+  }) : super(key: key);
+
+  final Widget child;
+  final FocusNode? focusNode;
+
+  @override
+  State<_FocusTrap> createState() => _FocusTrapState();
+}
+
+class _FocusTrapState extends State<_FocusTrap> {
+  final Map<Type, GestureRecognizerFactory> _gestures = <Type, GestureRecognizerFactory>{};
+  late FocusNode _focusNode;
+
+  FocusNode get focusNode => widget.focusNode ?? _focusNode;
+
+  void _stealFocus(TapDownDetails details) {
+    if (details.kind != PointerDeviceKind.mouse)
```

Comment:
do you know what is the native behavior for mobile if you have an external mouse?

Reply:
i don't have access to the hardware to test this. we could definitely just disable the behavior on iOS/android until we could test it

Reply:
(disabled for non-desktop platforms for now)